content/6_Advanced/BCC_2CC.mdx

```diff
@@ -25,7 +25,75 @@ frequency: 1
 
 <Problems problems="sam2" />
 
-(implementation)
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+const int MAX_N = 3e5;
+
+int timer;
+int scc;
+int id[MAX_N];
+int low[MAX_N];
+vector<int> g[MAX_N];
+vector<int> two_edge_components[MAX_N];
+stack<int> st;
```

Comment:
rename `g` to `neighbors`
what does `low` do? better name and/or comment
what does `st` do? better name and/or comment

Reply:
Done. st is a `stack` I think it's pretty obvious what it does.

Reply:
to *you* it’s obvious.
what does the stack keep track of? how is it used?

Reply:
unresolved.

Reply:
Ok, I added an explanation